federated_aws_cli/login.py

```diff
@@ -337,21 +340,29 @@ def aws_federate(self):
             "sessionToken": self.credentials["SessionToken"],
         }
 
-        params = urlencode({
+        query = urlencode({
             "Action": "getSigninToken",
             "Session": json.dumps(creds),
         })
+        logger.debug("Web Console params: {}".format(query))
 
-        logger.debug("Web Console params: {}".format(params))
-
-        url = "https://signin.aws.amazon.com/federation?Action=getSigninToken"
-        url += "&Session={}".format(quote_plus(json.dumps(creds)))
-
+        url_tuple = urlparse('https://signin.aws.amazon.com/federation')
+        url = urlunparse(url_tuple._replace(query=query))
         token = requests.get(url).json()
 
-        url = "https://signin.aws.amazon.com/federation?Action=login"
-        url += "&Destination=" + quote_plus("https://console.aws.amazon.com/")
-        url += "&SigninToken=" + token["SigninToken"]
+        account_id = self.role_arn.split(":")[4]
+        role = self.role_arn.split(':')[5].split('/')[-1]
+        issuer_url_query = urlencode({"account": account_id, "role": role})
+        issuer_url = urlunparse(
+            ('https', self.issuer_domain, '/', '', issuer_url_query, ''))
+
+        query = urlencode({
+            "Action": "login",
+            "Destination": "https://console.aws.amazon.com/",
+            "SigninToken": token["SigninToken"],
+            "Issuer": issuer_url
```

Comment:
Trailing comma.  :)

Reply:
Fixed in d12afa1854849ffbb40cf1c2a7e77458ccf7ce54